test/e2e/persistentips_test.go

```diff
@@ -311,3 +339,7 @@ func removeFinalizersPatch() ([]byte, error) {
 	}
 	return json.Marshal(patch)
 }
+
+func getIPsFromVMIStatus(vmi *kubevirtv1.VirtualMachineInstance) ([]string, error) {
```

Comment:
Since the same test portfolio is run for both secondary/primary UDN, we could either copy the entire tests for primary UDN (making a lot of duplicate code), OR try to join them. The main change is the way both tests retrieve the IPs from the VMI:
For secondary: retrieve the IPs from the VMI.Status
For primary UDN: retrieve the IPs from the pod's network-status annotation. 

So right now you're right - it's seems like a unneeded generalization of the function. 
But what we eventually need is a general `ipsFrom` function, that will satisfy both options.

BTW once the IPs could be retrieved from the VMI.status also for primary UDNs then we could remove this complexity.
I will adjust the commit message to better explain this.

Reply:
I still don't get it. 

Why does the function's signature return an error ? 

Will a follow up commit change this to throw an error ? If so, why don't you update the function's signature to return an error when it is needed ?

Reply:
ACK, I'll introduce the error return param in the relevant commit.

Reply:
DONE